databricks-sdk-java/src/main/java/com/databricks/sdk/core/ProxyConfig.java

```diff
@@ -0,0 +1,72 @@
+package com.databricks.sdk.core;
+
+public class ProxyConfig {
+  private String host;
+  private int port;
+  private String username;
+  private String password;
+  private ProxyAuthType proxyAuthType;
+  private boolean useSystemProperties;
```

Comment:
Should this default to `true`? If users want to use system properties, it seems strange that they would need to set it explicitly.

If we're introducing this, we should also use it to control whether `.useSystemProperties()` is called on line 75.

Reply:
Not sure about this one, I felt like my app should be independent of system proxy settings unless i choose that explicitly but I understand the other way could also feel natural.